measurement_parser.py

```diff
@@ -0,0 +1,102 @@
+    ## Certain characters that need to get filtered out from ingredient
+    ingredient = ingredient.strip().replace(',', '').replace('- ', ' ').replace('\xa0', ' ').replace(';', '').replace('"', '').replace('(', '').replace(')', '').replace('%', '').replace(' to ', '-').replace('½', ' 1/2').replace('¼', ' 1/4').replace('¾', ' 3/4').replace('*', '')
+    ## Handle parentheses later
+    singular_phrase = []
+    for word in ingredient.split(' '):
+        word = word.strip().lower()
+        if '-' in word and len(word) > 1:
+            first, second = word.split('-')[0], word.split('-')[1]
+            ## Some recipes list amounts as 1-2, 3-4, etc., we decided for this that we will take the smaller of the two numbers
+            if first.isnumeric() and second.isnumeric():
+                word = first
+            ## Other case would be a situation such as 8-ounce, for this we just need to split the two apart
+            elif first.isnumeric():
+                singular_phrase.append(first)
+                word = second
+            else:
+                word = word.replace('-', ' ')
+            ## Handles if amount is a decimal ex. .01 ounce
+        if '.' in word:
+            if word.endswith('.'):
+                word = word.replace('.', '')
+            ##else:
+                ##with open('raw_data/unknown_ingredients.txt', 'a+') as f:
+                    ##f.write('unknown' + ingredient + '\n')
+        if '/' in word:
+            if word.endswith('/'):
+                word = word.replace('/', '')
+            ## Weird edgecase when recipe author has terrible formating, wrote ingredient is 14/-inch
+            if len(word.split(' ')) > 1:
+                if word.split(' ')[0].endswith('/'):
+                    word = word.replace('/', '')
+
+        if word != '' and word != 'tbs':
+            singular = p.singular_noun(word)
+        else:
+            singular = False
+
+        if singular is not False:
+            singular_phrase.append(singular)
+        else:
+            if word in word_numbers:
+                singular_phrase.append(word_to_int[word])
+            else:
+                singular_phrase.append(word)
+
+    return " ".join(singular_phrase).lower()
+
+def get_measurements(ingredient):
+    matches = []
+
+    ## Handles case where ingredient specifies cutting into a certain quantity ex. 1/2 inch cube, do not want to accidently link this quantity to an ingredient
+    if 'cut into' in ingredient:
+        ingredient_split = ingredient.split('cut into')[0]
+
+    ingredient_split = ingredient.split(' ')
+    ## Remove extra spaces
+    while '' in ingredient_split:
+        ingredient_split.remove('')
+
+    counter = 0
+
+    while counter < len(ingredient_split):
+        if (ingredient_split[counter].isnumeric() or (ingredient_split[counter][0].isnumeric() and '/' in ingredient_split[counter] and len(ingredient_split[counter]) == 3)) and counter < len(ingredient_split) - 1:
```

Comment:
over here you cal 'ingredient_split[counter]' a lot, this could probably be simplified by making it a variable and then just calling the variable